hazelcast/src/test/java/com/hazelcast/util/scheduler/SecondsBasedEntryTaskSchedulerTest.java

```diff
@@ -32,151 +34,151 @@
     private TaskScheduler executorService = mock(TaskScheduler.class);
 
     @Mock
+    @SuppressWarnings("unchecked")
     private ScheduledEntryProcessor<Integer, Integer> entryProcessor = mock(ScheduledEntryProcessor.class);
 
+    private SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler;
+
     @Before
+    @SuppressWarnings("unchecked")
     public void mockScheduleMethod() {
-        when(executorService.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class))).thenReturn(
-                mock(ScheduledFuture.class));
+        when(executorService.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class)))
+                .thenReturn(mock(ScheduledFuture.class));
     }
 
     @Test
     public void test_scheduleEntry_postpone() {
-        final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
+        scheduler = new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
 
-        scheduler.schedule(100, 1, 1);
+        assertTrue(scheduler.schedule(100, 1, 1));
         assertNotNull(scheduler.get(1));
         assertEquals(1, scheduler.size());
     }
 
     @Test
     public void test_rescheduleEntry_postpone() {
-        final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
+        scheduler = new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
 
-        scheduler.schedule(100, 1, 1);
-        scheduler.schedule(10000, 1, 1);
+        assertTrue(scheduler.schedule(100, 1, 1));
+        assertTrue(scheduler.schedule(10000, 1, 1));
         assertNotNull(scheduler.get(1));
         assertEquals(1, scheduler.size());
     }
 
     @Test
-    public void test_dontRescheduleEntryWithinSameSecond_postpone() {
-        System.setProperty(ClockProperties.HAZELCAST_CLOCK_IMPL, StaticClock.class.getName());
-        try {
-            final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                    new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
-
-            scheduler.schedule(0, 1, 1);
-            scheduler.schedule(0, 1, 1);
-            assertNotNull(scheduler.get(1));
-            assertEquals(1, scheduler.size());
-        } finally {
-            System.clearProperty(ClockProperties.HAZELCAST_CLOCK_IMPL);
+    public void test_doNotRescheduleEntryWithinSameSecond_postpone() {
```

Comment:
:+1:

On Wed, Jul 20, 2016 at 3:28 PM, Donnerbart <notifications@github.com>
wrote:

> In
> hazelcast/src/test/java/com/hazelcast/util/scheduler/SecondsBasedEntryTaskSchedulerTest.java
> <https://github.com/hazelcast/hazelcast/pull/8544#discussion_r71523514>:
>
> >          assertNotNull(scheduler.get(1));
> >          assertEquals(1, scheduler.size());
> >      }
> >
> >      @Test
> > -    public void test_dontRescheduleEntryWithinSameSecond_postpone() {
> > -        System.setProperty(ClockProperties.HAZELCAST_CLOCK_IMPL, StaticClock.class.getName());
> > -        try {
> > -            final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
> > -                    new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
> > -
> > -            scheduler.schedule(0, 1, 1);
> > -            scheduler.schedule(0, 1, 1);
> > -            assertNotNull(scheduler.get(1));
> > -            assertEquals(1, scheduler.size());
> > -        } finally {
> > -            System.clearProperty(ClockProperties.HAZELCAST_CLOCK_IMPL);
> > +    public void test_doNotRescheduleEntryWithinSameSecond_postpone() {
>
> I implemented the loop and gave the test a timeout, so we don't have to
> deal with a max retry for the loop. The whole test class finishes in under
> 50 ms on my machine, so 10 seconds should be a very generous timeout value
> for this method.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hazelcast/hazelcast/pull/8544/files/72db4fc600546f794047ae02697ee71d8d4cd80f#r71523514>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfJRkAYiHb6EYOKCUxjfAkObHXylIdhks5qXiKJgaJpZM4JO9pG>
> .
>